hpy/devel/include/hpy/hpytype.h

```diff
@@ -8,36 +8,45 @@
 #define HPyAPI_UNUSED static inline
 #endif /* __GNUC__ */
 
+typedef enum {
+    HPyType_BuiltinShape_Legacy = -1,
+    HPyType_BuiltinShape_Object = 0,
+    HPyType_BuiltinShape_Type = 1,
+    HPyType_BuiltinShape_Long = 2,
+    HPyType_BuiltinShape_Float = 3,
+    HPyType_BuiltinShape_Unicode = 4,
+    HPyType_BuiltinShape_Tuple = 5,
+    HPyType_BuiltinShape_List = 6,
+} HPyType_BuiltinShape;
+
 typedef struct {
     const char* name;
     int basicsize;
     int itemsize;
     unsigned long flags;
     /*
-       A type whose struct starts with PyObject_HEAD is a legacy type. A
-       legacy type must set .legacy = true in its HPyType_Spec.
-       A type is a non-legacy type, also called HPy pure type, if its struct
-       does not include PyObject_HEAD. Using pure types should be preferred.
-       Legacy types are available to allow gradual porting of existing CPython
-       extensions.
-
-       A type with .legacy_slots not NULL is required to have .legacy = true and to
-       include PyObject_HEAD at the start of its struct. It would be easy to
-       relax this requirement on CPython (where the PyObject_HEAD fields are
-       always present) but a large burden on other implementations (e.g. PyPy,
-       GraalPython) where a struct starting with PyObject_HEAD might not exist.
+       A type whose struct starts with PyObject_HEAD is a legacy type. A legacy
+       type must set .builtin_shape = HPyType_BuiltinShape_Legacy in its
```

Comment:
`A type whose struct starts with PyObject_HEAD is a legacy type.`

this is not strictly correct. I think that after this PR we can write something like this, can't we:
```c
struct MyMetaClass {
    PyTypeObject base;
    ...
};
```
i.e. it is a legacy type even if it doesn't literally start with `PyObject_HEAD`.
One quick fix could be "a type whose struct starts with PyObject_HEAD or equivalent is a legacy type"?
Do we have any better wording?

Reply:
done